### DISCUSSION TITLE
need some help to communicate correctly with backend

### DISCUSSION THREAD

**Discussion by zhoujingfighting** [eclipse-theia/theia#9604] [Q&A] [answered]
hi , i'm comming again , this time i want to ask something  about JSON rpc communication,
here is my [code repo](https://github.com/zhoujingfighting/theia-Json-rpc/tree/main) 

all my code is in the folder **communication**,when i press the command , nothing showed in the console,and no error was reported

I will appreciate for your help!!

thanks!!

**Comment by vince-fugnitto** [comment, marked as answer]
> nothing showed in the console

@zhoujingfighting the log is present on the backend, where the server is started:

```
yarn run v1.22.4
$ theia start
root INFO Theia app listening on http://localhost:3000.
root INFO Configuration directory URI: 'file:///home/vince/.theia'
root INFO Changed application state from 'init' to 'started_contributions'.
root INFO Changed application state from 'started_contributions' to 'attached_shell'.
root INFO >>> Restoring the layout state...
root INFO <<< The layout has been successfully restored.
root INFO Changed application state from 'attached_shell' to 'initialized_layout'.
root INFO Changed application state from 'initialized_layout' to 'ready'.
root INFO start
root INFO Helloworld
```

**Reply by zhoujingfighting** [reply]
@vince-fugnitto  oh , I didn't see that at the VScode terminal that time , maybe I should check out the output tommorrow , anyway , thanks ,  besides , is there anything wrong with my codes ? my coding logic is right ?

**Reply by vince-fugnitto** [reply]
@zhoujingfighting I haven't verified the code, only confirmed that the command works and output is displayed
.

**Reply by zhoujingfighting** [reply]
@vince-fugnitto ,oh , anyway ,thanks ,vince  , I saw that output at the VScode terminal

**Reply by zhoujingfighting** [reply]
@vince-fugnitto maybe I did not refresh the page under the watch , the program code changed but the page did not
